### PR TITLE
content_haven_on_demand

### PR DESCRIPTION
**Pull request by dashapaster**
Created flow witch use HPE Haven onDemand APi

### CODE REVIEW

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          do:
+            rest.http_client_get:
+               - url: ${str(url) + str(jobID) + "?apikey=" + str(apikey)}
+               - proxy_host: proxy.houston.hp.com
+               - proxy_port: '8080'
+          publish:
+             - error_message
+             - return_result
+             - return_code
+          navigate:
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster  navigate should be: - SUCCESS: ... - FAILURE: .... according with the latest updates

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     - on_failure:
+            - print_fail:
+                  do:
+                    base.print_text:
+                        - text: "${error_message}"
+  outputs:
+      - result: ${return_result}
+      - resultOfRecogn: ${value if error_message=='' else 0}
+  results:
+    - SUCCESS: ${error_message==''}
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster  should not use validation expression on results if is a flow... according with the latest updates

File: content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          do:
+            speechRecognision:
+               - url
+               - video
+               - apikey
+
+          publish:
+             - jobID
+             - error_message
+          navigate:
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl]: @dashapaster  ... same here as first observation

File: content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+           do:
+             checkStatus:
+               - url: ${urlForResult}
+               - jobID: ${jobID}
+               - apikey: ${apikey}
+
+           publish:
+             - error_message
+             - result: ${resultOfRecogn}
+           navigate:
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl]: @dashapaster  ... same here as first observation

File: content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            - proxy_port: '8080'
+            - method: POST
+            - multipart_bodies: ${"apikey=" + str(apikey)}
+            - multipart_files: ${video}
+
+        publish:
+            - error_message
+            - return_result
+            - return_code
+        navigate:
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl]: @dashapaster  ... same here as first observation

File: content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+           - error_message
+    - on_failure:
+          - print_fail:
+                do:
+                  base.print_text:
+                    - text: "${error_message}"
+  outputs:
+      - jobID: ${value if error_message=='' else 0}
+  results:
+     - SUCCESS: ${error_message==''}
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl]: @dashapaster  ... same here as second observation

File: content/io/cloudslang/haven_onDemand/p.prop.sl [hunk 7]
```diff
@@ -0,0 +1,9 @@
+namespace: io.cloudslang.haven_onDemand
+
+properties:
+  apikey: "c0e7aae8-f657-4f00-aad4-f839f956800e"
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/p.prop.sl]: @dashapaster ... don't think is a good ideea to put sensitive data as public ones.

**Comment by mihaitusa** [comment]
@dashapaster  Pease handle comments ... merge the master into your branch... rebuild

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 8]
```diff
@@ -0,0 +1,74 @@
+
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster  Probably is better to rename the file to be aligned with the rest of the content... I mean snake_case

File: content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl [hunk 9]
```diff
@@ -0,0 +1,96 @@
+
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/sendVoiceRecToMail.sl]: @dashapaster  Probably is better to rename the file to be aligned with the rest of the content... I mean snake_case

File: content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl [hunk 10]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/speechRecognision.sl]: @dashapaster  Probably is better to rename the file to be aligned with the rest of the content... I mean snake_case

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+#!!
+#! @description: Gets results of Speech Recognision, with was made by HPE Haven OnDemand API.
+#! @input url: URL to Speech Recognision API
+#! @input apikey: user's API Keys
+#! @input jobID: name of request, witch is returned by havenondemand.com
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster The inputs should be snake_case to be aligned with the rest of the content

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+  rest: io.cloudslang.base.network.rest
+  json: io.cloudslang.base.json
+  file: io.cloudslang.base.files
+  base: io.cloudslang.base.print
+flow:
+  name: checkStatus
+
+  inputs:
+    - url
+    - jobID
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster The inputs should be snake_case to be aligned with the rest of the content

File: content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+          - error_message
+
+     - on_failure:
+            - print_fail:
+                  do:
+                    base.print_text:
+                        - text: "${error_message}"
+  outputs:
+      - result: ${return_result}
+      - resultOfRecogn: ${value if error_message=='' else 0}
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/voice_recognision/checkStatus.sl]: @dashapaster The outputs should be snake_case to be aligned with the rest of the content

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+            rest.http_client_get:
+               - url: ${str(speechResultApi) + str(jobID) + "?apikey=" + str(apikey)}
+
+          publish:
+             - error_message
+             - return_result
+             - return_code
+          navigate:
+             - SUCCESS: get_status_recognision
+             - FAILURE: checkStatus
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: @dashapaster  Isn't suppose to failure here? You want to made a loop? What happens if the machine is down and the API is not reachable ?

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+     - get_status_recognision:
+            do:
+              json.get_value:
+                - json_input: ${return_result}
+                - json_path: ['status']
+            publish:
+              - status: ${value}
+              - error_message
+
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: @dashapaster  You have a several get steps in here ... I think is better to use navigate on each step and to choose a meaningful failure label if something goes wrong. What if in this step you get an "500" internal error for e.g.? How do I know as a user if something went wrong on this step?

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+     - get_result_recognision:
+        do:
+          json.get_value:
+            - json_input: ${return_result}
+            - json_path: ['actions',0,'result','document',0,'content']
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: @dashapaster  Same as above... you cannot be that sure that you will always find something in response JSON that matches the path ['actions',0,'result','document',0,'content']... so, I think is better to have navigate here as well...

File: test/io/cloudslang/haven_onDemand/speech_recognision/sendTranscriptToMail.inputs.yaml [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    - file: <file>
+    - speechResultApi: "https://api.havenondemand.com/1/job/result/"
+    - hostname: <hostname>
+    - port: <port>
+    - from: <email>
+    - to: <email>
+  description: Test that sendVoiceRecToMail.sl finished with result = SUCCESS
+  testFlowPath: io.cloudslang.haven_onDemand.voice_recognision.sendTranscriptToMail
+  result: SUCCESS
+  testSuites: [haven]
```
Comment (mihaitusa) [inline test/io/cloudslang/haven_onDemand/speech_recognision/sendTranscriptToMail.inputs.yaml]: @dashapaster  [haven-local] in here... you don't actually run a build test in here.

**Comment by mihaitusa** [comment]
@dashapaster  Please handle the comments.

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl (hunk 14 above)
Comment (dashapaster) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: yes, i want to made like 'while' loop

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl (hunk 16 above)
Comment (dashapaster) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: @mihaitusa  but it is the last step of flow, and if it fail - there is on_failure...

File: content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl (hunk 15 above)
Comment (dashapaster) [inline content/io/cloudslang/haven_onDemand/speech_recognision/checkStatus.sl]: @mihaitusa do you think it is better to create somethink like print_fail just for this step?

**Pull request merged**